Debugger/tests/Unit/AgentTest.php

```diff
@@ -112,7 +112,111 @@ public function testDaemonOptions()
+            $variable = $stackframes[0]->locals()[0];
+            $this->assertEquals(500, strlen($variable->info()['value']));
+            return true;
+        };
+        $agent = $this->setupAgent($itemMatcher, []);
+        $locals = [[
+            'name' => 'longString',
+            'value' => str_repeat('a', 10000)
+        ]];
+        $agent->handleSnapshot([
+            'id' => 'snapshot1',
+            'evaluatedExpressions' => [],
+            'stackframes' => [
+                ['filename' => 'file1.php', 'line' => 20, 'locals' => $locals]
+            ]
+        ]);
+    }
+
+    public function testSetsMaxStringLength()
+    {
+        $itemMatcher = function ($item) {
+            $this->assertEquals('debuggeeId', $item[0]);
+            $this->assertInstanceOf(Breakpoint::class, $item[1]);
+            $stackframes = $item[1]->stackFrames();
+            $this->assertCount(1, $stackframes);
+            $this->assertEquals(1, count($stackframes[0]->locals()));
+            $variable = $stackframes[0]->locals()[0];
+            $this->assertEquals(1000, strlen($variable->info()['value']));
+            return true;
+        };
+        $agent = $this->setupAgent($itemMatcher, [
+            'maxValueLength' => 1000
+        ]);
+        $locals = [[
+            'name' => 'longString',
+            'value' => str_repeat('a', 10000)
+        ]];
+        $agent->handleSnapshot([
+            'id' => 'snapshot1',
+            'evaluatedExpressions' => [],
+            'stackframes' => [
+                ['filename' => 'file1.php', 'line' => 20, 'locals' => $locals]
+            ]
+        ]);
+    }
+
+    public function testDefaultMaxPayloadSize()
+    {
+        $itemMatcher = function ($item) {
+            $this->assertEquals('debuggeeId', $item[0]);
+            $this->assertInstanceOf(Breakpoint::class, $item[1]);
+            $stackframes = $item[1]->stackFrames();
+            $this->assertCount(1, $stackframes);
+            $this->assertEquals(245, count($stackframes[0]->locals()));
+            return true;
+        };
+        $agent = $this->setupAgent($itemMatcher, []);
+        $locals = [];
+        for ($i = 0; $i < 1000; $i++) {
+            $locals[] = ['name' => 'var' . $i, 'value' => str_repeat('.', $i)];
+        }
+        $agent->handleSnapshot([
+            'id' => 'snapshot1',
+            'evaluatedExpressions' => [],
+            'stackframes' => [
+                ['filename' => 'file1.php', 'line' => 20, 'locals' => $locals]
+            ]
+        ]);
+    }
+
+    public function testSetsMaxPayloadSize()
+    {
+        $itemMatcher = function ($item) {
+            $this->assertEquals('debuggeeId', $item[0]);
+            $this->assertInstanceOf(Breakpoint::class, $item[1]);
+            $stackframes = $item[1]->stackFrames();
+            $this->assertCount(1, $stackframes);
+            $this->assertEquals(34, count($stackframes[0]->locals()));
```

Comment:
How did you calculate `34`?

Reply:
The payload size of entry `$i` is `6 + $i + 3 + floor(log10($i))`. The sum of that from 0 to 34 is 925 which is the first element over 900 (we leave a ~100 byte buffer for extra fields). I'll add a comment or make the formula simpler.

Reply:
Simplified this and added a comment.